src/main/scala/com/ambiata/disorder/EncodingString.scala

```diff
@@ -0,0 +1,23 @@
+package com.ambiata.disorder
+
+import org.scalacheck._, Arbitrary._, Gen._
+import scala.io.Codec
+import scalaz._, Scalaz._
+import Arbitraries._
+import GenPlus._
+
+/**
+  Produce a string which is valid for the associated codec
+  */
+case class EncodingString(value: String, codec: Codec)
+
+object EncodingString {
+  implicit def EncodingStringArbitrary: Arbitrary[EncodingString] =
+    Arbitrary(checkGenTuple[EncodingString, S, Codec](
+        (s, c) => EncodingString(s.value, c)
+      , "EncodingString"
+      , es => validForCodec(es.value, es.codec)))
+
+  def validForCodec(s: String, c: Codec): Boolean =
```

Comment:
> The only reason to do one over the other is performance IMO, the currently implemented solution avoids the pitfalls mentioned in the link.

Yeah was just thinking out loud really. If performance isn't an issue with what we have is certainly much easier.